File: 1.Data structure/3.Stack/크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = [0]
    cnt = 0
    for i in moves:
        for j in range(0,len(board)):
            if board[j][i-1] != 0:
                if board[j][i-1] == answer[len(answer)-1]:
                    answer.pop()
                    cnt += 2
                else:
                    answer.append(board[j][i-1])
                board[j][i-1] = 0
                break
    return cnt

File: 1.Data structure/3.Stack/크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        Stack<Integer> stack = new Stack<>();
        int answer = 0;
        for (int i : moves){
            int k = 0;
            while (board[k][i-1] == 0 && k < board.length-1){
                k += 1;
            }
            if (!stack.empty() && stack.peek() == board[k][i-1]){
                stack.pop();
                answer += 2;
            }else if (board[k][i-1] != 0){
                stack.push(board[k][i-1]);
            }
            board[k][i-1] = 0;
        }
        return answer;
    }
}